### PR TITLE
EH custom endpoint support

### PR DESCRIPTION
**Pull request by yunhaoling**
add support for the Event Hub Issue https://github.com/Azure/azure-sdk-for-python/issues/16047.

eventhub PR: https://github.com/Azure/azure-sdk-for-python/pull/16295/files

I intend to hide "hostname" from the "auth" API surface first, as it's not part of the AMQP spec for a feature for EH only

### CODE REVIEW

**Review by annatisch — APPROVED** [review]
LGTM - maybe add changelog notes?

**Comment by yunhaoling** [comment]
@annatisch , I'm a bit torn on mentioning it the release note as it's not part of the core amqp spec and it's there to support eventhub.

apart from that, do you think the naming of the parameter should be more explicit such as `custom_hostname`, I'm afraid that the `hostname` is too general.

File: uamqp/authentication/cbs_auth.py [hunk 1]
```diff
@@ -223,21 +223,22 @@ def __init__(self, audience, uri, token,
...
+                 encoding='UTF-8',
+                 **kwargs):  # pylint: disable=no-member
         self._retry_policy = retry_policy
         self._encoding = encoding
         self.uri = uri
         parsed = compat.urlparse(uri)  # pylint: disable=no-member
 
         self.cert_file = verify
-        self.hostname = parsed.hostname.encode(self._encoding)
+        self.hostname = (kwargs.get("hostname") or parsed.hostname).encode(self._encoding)
```
Comment (yunhaoling) [inline uamqp/authentication/cbs_auth.py:241]: `hostname` rename to `custom_endpoint_hostname`

**Comment by yunhaoling** [comment]
conclusion: not exposing the config now, also renamed to `custom_endpoint_hostname` for clarity

**Pull request merged**